de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/MetadataTemplateViewDialog.java

```diff
@@ -100,45 +111,63 @@ public MetadataTemplateViewDialog(MetadataView.Presenter presenter,List<Avu> tem
 
                 avu.setValue(value);
             }
-
             avus.add(avu);
-        }
+
+        });
         return avus;
     }
-    
+
     public boolean isValid() {
-      List<IsField<?>> fields = FormPanelHelper.getFields(widget);
-            for (IsField<?> f : fields) {
-                if (!f.isValid(false)) {
-                    valid = false;
-                }
+        List<IsField<?>> fields = FormPanelHelper.getFields(widget);
+        for (IsField<?> f : fields) {
+            if (!f.isValid(false)) {
+                valid = false;
             }
+        }
 
-      return valid;
+        return valid;
     }
-    
+
     private void buildAvuMap() {
-    	for (MetadataTemplateAttribute attribute: attributes) {
-    		Avu md = findMetadataForAttribute(attribute.getName());
-    		templateAttrAvuMap.put(attribute.getName(),md);
-    	}
+        for (MetadataTemplateAttribute attribute : attributes) {
+            List<Avu> mds = findMetadataForAttribute(attribute.getName());
+            if (mds != null && !mds.isEmpty()) {
+                mds.forEach(md -> {
+                    String tag = getTagForMetadata(md);
+                    templateTagAvuMap.put(tag, md);
+                    templateTagAtrrMap.put(tag, attribute);
+                    // GWT.log(tag + "-->" + attribute.getName());
+                });
+            } else {
+                Avu avu = presenter.setAvuModelKey(MetadataPresenterImpl.newMetadata(attribute.getName(),
+                                                                                     "",
+                                                                                     "")); //$NON-NLS-1$ //$NON-NLS-2$
+                String tag = getTagForMetadata(avu);
+                templateTagAvuMap.put(tag, avu);
+                templateTagAtrrMap.put(tag, attribute);
+                // GWT.log(tag + "-->" + attribute.getName());
+            }
+
+        }
     }
-    
-    private Avu findMetadataForAttribute(String attribute) {
-    	if(templateMd != null) {
-	    	for(Avu md: templateMd) {
-	    		if(md.getAttribute().equals(attribute)) {
-	    			return md;
-	    		}
-	    	}
-    	}
-		return null;
+
+    private List<Avu> findMetadataForAttribute(String attribute) {
```

Comment:
That's fair.  It's not at all critical for this PR, but I think I was leaning towards kind of what we do with the liststores. The presenters (for the most part) manipulate the data in the liststores so the view only has to worry about displaying the information, not manipulating it.  So in this case, I was envisioning the presenter would create and then pass in the FastMaps to the view.  Then for `getMetadataFromTemplate`, instead of the presenter calling that method which then has to call the `newMetadata()` method from the presenter and then returning the value back, it could all be in the presenter for example.  
Again, not critical for this PR, and I do see your point about wanting to keep view-related things in the view.